es6-renderer.js

```diff
@@ -11,7 +11,21 @@ const getPartial = (path, cb = 'resolveNeutral') => {
   };
   return new Promise(findFile);
 };
-    
+
+const defineAllVariables = function(template, locals) {
+  const localsDef = locals;
+  const pattern = /\$\{[\s]*?([\w\d]+)[\s]*?[\}\?]/gm;
```

Comment:
The complexity of the pattern might have some serious performance issues

Reply:
I think this function is trying to add properties to locals for each variable it encounters in the template that is not in locals.

In terms of optimizations. I think the regex can be simpler if you want to deal with normal templates. You don't want to create locals for ill-typed template variables. Also, you don't need the `m` flag, just the `g` flag.

Is it possible to let the es-renderer do this optionally? And if you make this optional, you could even make the kind of pattern a configuration. I don't have time to help you with patterns, but it's something that you can test of off es-renderer in some benchmark on https://jsperf.com/